src/utils/Deprecated.js

```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const Deprecated = ({componentName, message, children}) => {
+  console.warn(`WARNING! ${componentName} will be deprecated soon. ${message}`)
+  return children
+}
+
```

Comment:
I really really like this! Thoughts about adding a `since` prop that specifies the version that the deprecation was created in? Could also be useful for us to grep through the codebase and see which deprecations are ready to be removed.

Reply:
yeah that's a great idea! I'll add it!

Reply:
I added a `version` prop instead, for which version the component is planning to be deprecated in. I'm assuming the term `deprecated` to mean `completely removed` so this is what I was thinking the process would be:

1. When you are getting ready to remove a component or prop, first wrap the component in `Deprecated` and specify which future version the component will be completely removed in. 

If there is a migration strategy (in this case it's to use `H1`, etc instead), that new API should be available when the deprecation notice is added so the users can start migrating immediately. In other words, in most cases when you are adding a new component or prop that replaces an old component/prop, keep _both_ around for a period of time to allow folks to migrate. It might not always make sense to do this. I could see situations where we want to remove a component completely without a replacement (for instance, removing StyledOcticon in favor of the external @primer/styled-octicons library).

2. Keep the deprecation warning + old component around for a specified period of time. In this case, since we're going to ship 19.0.0 very soon, I set it for 20.0.0 so that we can give folks time to see the warning.

3. When shipping a major release, search the codebase for `version=<VERSION ABOUT TO BE RELEASED>`. Completely remove the components wrapped in `Deprecated` and all related type definitions, exports, docs, tests etc.

What do you think?

Reply:
I think "deprecated" is commonly used as "still available but in the process of being changed or removed;" if an API is completely removed, it's no longer deprecated.

That said, I'm okay with using a prop to note which version we plan on removing the feature in. I guess it's not all that important to know when the API was deprecated. The only downside is that time between major releases is not always consistent.

👍 on `DeprectionWarning` as a name